### ISSUE TITLE
Invalid characters in HTTP requests can cause NullPointerException in the access logs

### ISSUE THREAD

**Issue by swcm-mpilz** [akkinoc/logback-access-spring-boot-starter#139]
## Describe the bug

During a recent attack attempt on one of our services, we noticed that while Tomcat correctly rejects invalid characters in a request with a 400 Bad Request, it also forwards the request to the access logger, which itself throws a NullPointerException. This seems to be caused by the fact that any error in the tomcat parsing pipeline results in a request with a null `requestURI`. 

Stack trace:
```
java.lang.NullPointerException: <get-requestURI>(...) must not be null
	at dev.akkinoc.spring.boot.logback.access.tomcat.LogbackAccessTomcatEventSource.getRequestURI(LogbackAccessTomcatEventSource.kt:85)
	at dev.akkinoc.spring.boot.logback.access.tomcat.LogbackAccessTomcatEventSource$requestURL$2.invoke(LogbackAccessTomcatEventSource.kt:94)
	at dev.akkinoc.spring.boot.logback.access.tomcat.LogbackAccessTomcatEventSource$requestURL$2.invoke(LogbackAccessTomcatEventSource.kt:93)
	at kotlin.UnsafeLazyImpl.getValue(Lazy.kt:81)
	at dev.akkinoc.spring.boot.logback.access.tomcat.LogbackAccessTomcatEventSource.getRequestURL(LogbackAccessTomcatEventSource.kt:93)
	at dev.akkinoc.spring.boot.logback.access.LogbackAccessEvent.getRequestURL(LogbackAccessEvent.kt:94)
	at ch.qos.logback.access.net.URLEvaluator.evaluate(URLEvaluator.java:47)
	at ch.qos.logback.core.filter.EvaluatorFilter.decide(EvaluatorFilter.java:65)
	at ch.qos.logback.core.spi.FilterAttachableImpl.getFilterChainDecision(FilterAttachableImpl.java:57)
	at ch.qos.logback.access.spi.AccessContext.getFilterChainDecision(AccessContext.java:95)
	at dev.akkinoc.spring.boot.logback.access.LogbackAccessContext.emit(LogbackAccessContext.kt:62)
	at dev.akkinoc.spring.boot.logback.access.tomcat.LogbackAccessTomcatValve.log(LogbackAccessTomcatValve.kt:75)
```

The fact that Tomcat calls the access logger on invalid requests seems to be very much intended (see documentation of [Container.java#logAccess](https://github.com/apache/tomcat/blob/49b5aac59a5db551383868ccd66f3e5513fe109a/java/org/apache/catalina/Container.java#L473) ), so the `LogbackAccessTomcatEventSource` should probably be able to handle null `requestURI`s.

## To Reproduce

- Setup minimal Spring server with logback-access-spring-boot-starter
- Send a request with an invalid URL like `curl 'http://localhost:8080/?a[]='`
- Observe expected Tomcat error: 
```
2022-03-17 15:38:00.605  INFO 255326 --- [nio-8080-exec-1] o.apache.coyote.http11.Http11Processor   : Error parsing HTTP request header
 Note: further occurrences of HTTP request parsing errors will be logged at DEBUG level.

java.lang.IllegalArgumentException: Invalid character found in the request target [/?a[]= ]. The valid characters are defined in RFC 7230 and RFC 3986
	at org.apache.coyote.http11.Http11InputBuffer.parseRequestLine(Http11InputBuffer.java:494)
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:269)
        <snip>
```
- Instead of access logs, a `NullPointerException` can be observed in the logs:

```
2022-03-17 15:38:00.619 ERROR 255326 --- [nio-8080-exec-1] o.apache.coyote.http11.Http11Processor   : Error processing request

java.lang.NullPointerException: <get-requestURI>(...) must not be null
	at dev.akkinoc.spring.boot.logback.access.tomcat.LogbackAccessTomcatEventSource.getRequestURI(LogbackAccessTomcatEventSource.kt:85)
	at dev.akkinoc.spring.boot.logback.access.tomcat.LogbackAccessTomcatEventSource$requestURL$2.invoke(LogbackAccessTomcatEventSource.kt:94)
       <snip>

2022-03-17 15:38:00.619  WARN 255326 --- [nio-8080-exec-1] o.a.catalina.connector.CoyoteAdapter     : Exception while attempting to add an entry to the access log

java.lang.NullPointerException: <get-requestURI>(...) must not be null
	at dev.akkinoc.spring.boot.logback.access.tomcat.LogbackAccessTomcatEventSource.getRequestURI(LogbackAccessTomcatEventSource.kt:85)
	at dev.akkinoc.spring.boot.logback.access.tomcat.LogbackAccessTomcatEventSource$requestURL$2.invoke(LogbackAccessTomcatEventSource.kt:94)
      <snip>
```

## Expected behavior

When an invalid request is sent to Tomcat, my expectation is that the access logger logs the parts that were successfully parsed and either does not log the invalid parts (in this case, the requestURI) or marks them as unparsable in some way.

### Environment

* **Version of this library used**: 3.0.0
* **Version of Java used**: 17

**Comment by akkinoc** [comment]
Thanks for report!
I fixed the bug in PR #140 and released it in v3.2.3.
Thanks.

**Comment by swcm-mpilz** [comment]
Amazing, thank you!
